File: app/src/main/java/org/mesonet/app/advisories/AdvisoryHeaderViewHolder.java
package org.mesonet.app.advisories;

import android.databinding.DataBindingUtil;
import android.view.LayoutInflater;
import android.view.ViewGroup;

import org.mesonet.app.R;
import org.mesonet.app.baseclasses.RecyclerViewHolder;
import org.mesonet.app.databinding.HeaderViewHolderBinding;
import org.mesonet.app.maps.MapsHeaderViewHolder;



public class AdvisoryHeaderViewHolder extends RecyclerViewHolder<AdvisoryModel.AdvisoryTypeModel, HeaderViewHolderBinding>
{
    public static AdvisoryHeaderViewHolder NewInstance(ViewGroup inParent)
    {
        return new AdvisoryHeaderViewHolder((HeaderViewHolderBinding) DataBindingUtil.inflate(LayoutInflater.from(inParent.getContext()), R.layout.header_view_holder, inParent, false));
    }
    public AdvisoryHeaderViewHolder(HeaderViewHolderBinding inBinding) {
        super(inBinding);
    }

    @Override
    public void SetData(AdvisoryModel.AdvisoryTypeModel inData) {
        if(inData != null)
        {
            HeaderViewHolderBinding binding = GetBinding();

            binding.headerText.setText(inData.mAdvisoryType + " " + inData.mAdvisoryLevel);
        }
    }
}

File: app/src/main/java/org/mesonet/app/advisories/AdvisoryHeaderViewHolder.kt
package org.mesonet.app.advisories

import android.databinding.DataBindingUtil
import android.databinding.ViewDataBinding
import android.view.LayoutInflater
import android.view.ViewGroup

import org.mesonet.app.R
import org.mesonet.app.baseclasses.RecyclerViewHolder
import org.mesonet.app.databinding.HeaderViewHolderBinding
import org.mesonet.dataprocessing.advisories.AdvisoryDisplayListBuilder


class AdvisoryHeaderViewHolder(inBinding: HeaderViewHolderBinding) : RecyclerViewHolder<Pair<AdvisoryDisplayListBuilder.AdvisoryDataType, AdvisoryDisplayListBuilder.AdvisoryData>, HeaderViewHolderBinding>(inBinding) {

    override fun SetData(inData: Pair<AdvisoryDisplayListBuilder.AdvisoryDataType, AdvisoryDisplayListBuilder.AdvisoryData>?) {

        val binding = GetBinding()

        if(binding != null)
            binding.headerText.text = (inData?.second)?.AdvisoryType()
    }

    companion object {
        fun NewInstance(inParent: ViewGroup): AdvisoryHeaderViewHolder {
            return AdvisoryHeaderViewHolder(DataBindingUtil.inflate<ViewDataBinding>(LayoutInflater.from(inParent.context), R.layout.header_view_holder, inParent, false) as HeaderViewHolderBinding)
        }
    }
}
